File: Backend/src/main/java/com/library/service/PublisherService.java
package com.library.service;

import java.util.List;

import com.library.entity.Publisher;

public interface PublisherService {

	List<Publisher> getAllPublishers();

	String addPublisherList(List<Publisher> publisherData);

	List<Publisher> searchPublisher(String text);

	Publisher addPublisher(Publisher publisherData);

	Publisher deletePublisher(int id);

	Publisher getPublisher(int id);

}

File: app/publisher.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { publisher } from './publisher';

@Injectable({
  providedIn: 'root'
})
export class PublisherService {
  deletePublisher(publisherId: any) {
    return this.httpClient.delete(`${this.baseURL}/library/publisher/${publisherId}`);
  }
  addPublisher(publisher: publisher) {
    return this.httpClient.post(`${this.baseURL}/library/publisher`,publisher);
  }
  getPublisherById(arg0: number) {
    return this.httpClient.get(`${this.baseURL}/library/publisher/${arg0}`);
  }
  searchPublisher(text:any) {
    return this.httpClient.get(`${this.baseURL}/library/publisher/search/${text}`)
  }
  getAllPublishers() {
    return this.httpClient.get(`${this.baseURL}/library/publisher`);
  }
  private baseURL = "http://localhost:8080";
  constructor(private httpClient : HttpClient){ }
}
